### DISCUSSION TITLE
Supabase setting callback url to explicit 127.0.01 instead of localhost which breaks azure oAuth on local development.

### DISCUSSION THREAD

**Discussion by asharrahman02** [supabase/supabase#27041] [Questions]
Hello, when I run supabase locally and specify the config file, my supabase instance will send the callback url as http://127.0.0.1:54321/auth/v1/callback instead of http://localhost:54321/auth/v1/callback. Although these mean the same thing, azure does not let me specify any http urls other than "localhost" which is causing issues for me. My callback and signin routes are identical to that in the documentation so I am not sure why this is occurring. Can anybody lend some insight or tell me what I am missing?

![Screenshot 2024-06-05 at 3 48 22 PM](https://github.com/supabase/supabase/assets/22228931/7c7ba962-1ed4-4efb-ba23-0f67742b39eb)

My config.toml:

[auth]
enabled = true
site_url = "http://localhost:3000"
additional_redirect_urls = ["http://localhost:3000", "http://localhost:54321/auth/v1/callback"]
jwt_expiry = 3600
enable_refresh_token_rotation = true
refresh_token_reuse_interval = 10
enable_signup = true
enable_anonymous_sign_ins = false
enable_manual_linking = false

...

[auth.external.azure]
enabled = true
client_id = "env(SUPABASE_AUTH_EXTERNAL_AZURE_CLIENT_ID)"
secret = "env(SUPABASE_AUTH_EXTERNAL_AZURE_SECRET)"
redirect_uri = "http://localhost:54321/auth/v1/callback"`

![image](https://github.com/supabase/supabase/assets/22228931/a78f2686-6ec9-4985-8ffc-c2b9735cfe33)

**Comment by vantanminh** [comment]
I have same problem

**Comment by avtarraikmo** [comment]
Did this issue ever get fixed? It still doesn't work for local testing due to Azure not supporting 127.0.0.1 for HTTP redirect. 

Either we need HTTPS redirect for 127.0.0.1 or we need to be able to send "localhost" in the url for Azure.
